virttest/env_process.py

```diff
@@ -908,10 +911,16 @@ def preprocess(test, params, env):
         params["image_raw_device"] = "yes"
 
     if params.get("storage_type") == "lvm":
-        lvmdev = qemu_storage.LVMdev(params, base_dir, "lvm")
-        params["image_name"] = lvmdev.setup()
-        params["image_raw_device"] = "yes"
-        env.register_lvmdev("lvm_%s" % params["main_vm"], lvmdev)
+        for image in params.objects("images"):
+            image_params = params.object_params(image)
+            if image_params.get("enable_lvm", "no") == "no":
+                continue
+            image_params["lv_name"] = lvm.get_lv_name(
+                image_params["image_name"], image_params["image_format"])
+            lvmdev = qemu_storage.LVMdev(image_params, base_dir, image)
```

Comment:
If it represents the entity of lvm pools, it should be initialized outside the loop, otherwise pools might be configured twice and that would result in errors.

The correct logic would be something like this:
```python
if params.get("storage_type") == "lvm":
    lvm_pools_initialize()

    ...

    for vm in vms:
        for image in vm.images:
            image_params = params.object_params(image)
            if image_params.get("enable_lvm", "no") == "no":
                continue
            # request volume
            vol = vol_lookup_by_image_params(image_params)
            ...
```